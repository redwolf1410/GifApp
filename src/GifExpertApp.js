import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
export const GifExpertApp = () => {

    const [categories,setCatergories] = useState(['Rick and Morty'])

    // const handleAdd = () =>{
    //     setCatergories([...categories, value])

    // }
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCatergories={setCatergories}/>  
            <hr />
            <ol>
                {
                    categories.map(category =>
                         <GifGrid 
                            key={category}
                            category={category}/>
                    )
                }

            </ol>
        </>

    )

}

