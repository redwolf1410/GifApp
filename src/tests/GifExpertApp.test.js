import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import React  from 'react';
import {GifExpertApp} from '../GifExpertApp'

describe('GifExpertApp tests', () => { 
    test('should show propperly', () => { 
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

     })
     test('should show a list of categories ', () => { 
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
         

      })

 })