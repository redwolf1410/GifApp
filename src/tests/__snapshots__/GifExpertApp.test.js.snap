// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests should show a list of categories  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp tests should show propperly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
  </ol>
</Fragment>
`;
